
#!/usr/bin/env python
PACKAGE = "CORC"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("virtual_mass_ratio", double_t, 0, "Virtual Mass Ratio", 0.8, 0.1, 1)
gen.add("desired_interaction_force", double_t, 1, "Desired Interaction Force", 0, -20, 20)
gen.add("m_admittance", double_t, 2, "Admittance Mass", 5, 1, 20)
gen.add("b_admittance", double_t, 3, "Admittance damper", 3, 0, 20)

interaction_enum = gen.enum([
                   gen.const("zero_torque", int_t, 1, "controller 1"),
                   gen.const("zero_velocity", int_t, 2, "controller 2"),
                   gen.const("ff_model_compensation", int_t, 3, "controller 3"),
                   gen.const("virtual_mass_controller", int_t, 4, "controller 4"),
                   gen.const("admittance_controller", int_t, 5, "controller 5"),
                   gen.const("param_est", int_t, 6, "controller 6"),
                   gen.const("chirp_torque", int_t, 7, "controller 7"),],
                   "Test Parameters Mode")

gen.add("controller_mode", int_t, 0, "Controller Mode", 2, 1, 7, edit_method=interaction_enum)

exit(gen.generate(PACKAGE, "CORC", "dynamic_params"))